Libs/GLib/Classes/MarshalHelper.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace GLib
+{
+    public static class MarshalHelper
+    {
+        public static K ToPtrAndFree<T, K>(T structure, Func<IntPtr, K> action)  where T : struct
+        {
+            K result;
+            IntPtr ptr = Marshal.AllocHGlobal(Marshal.SizeOf(structure));
+            try
+            {
+                Marshal.StructureToPtr(structure, ptr, false);
+                result = action(ptr);
+            }
```

Comment:
Since we're putting this in a try block, can we add some kind of logging if an exception is thrown?

e.g. maybe:

```suggestion
            }
            catch (Exception e)
            {
                Console.WriteLine($"Exception thrown while marshalling: {e.Message}");
            }
```

Reply:
Actually I don't want to catch the exception on our side of the code. But if the user is catching it the memory should be freed. This is why the try block is missing the catch part: 

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/try-finally

Do you think we should catch the exception?

Reply:
Ah, is the exception still propagated to the user then? As in, does finally run if the user catches the exception higher up the call stack? The docs don't make that very clear:

> Usually, when an unhandled exception ends an application, whether or not the finally block is run is not important. However, if you have statements in a finally block that must be run even in that situation, one solution is to add a catch block to the try-finally statement.

But then this implies that it does:

> Alternatively, you can catch the exception that might be thrown in the try block of a try-finally statement higher up the call stack. That is, you can catch the exception in the method that calls the method that contains the try-finally statement, or in the method that calls that method, or in any method in the call stack.

Reply:
Yep it should be like this, see [stackoverflow](https://stackoverflow.com/questions/9291437/use-a-try-finally-block-without-a-catch-block) which even states that this is similar to what a using block compiles to.